### DISCUSSION TITLE
Any love for tauri-egui?

### DISCUSSION THREAD

**Discussion by samkearney** [tauri-apps/tauri#10089] [General]
I think that integration of egui and Tauri enables a lot of powerful things, especially considering that the recent separation of windows and webviews might allow very flexible mixes of web-based and egui-based UI elements in a single application. But it appears that the relevant repos like [tauri-egui](https://github.com/tauri-apps/tauri-egui), the [egui fork](https://github.com/tauri-apps/egui) and [glutin_tao](https://github.com/tauri-apps/glutin) have been a bit neglected as of late.

I know the team is probably quite busy trying to get Tauri v2 over the finish line, but I'm curious if there are any plans to come back and update these plugins in the foreseeable future?

**Comment by FabianLars** [comment]
Please see the discussion here: https://github.com/tauri-apps/tauri-egui/issues/18 - The mentioned maintainer was needed somewhere else (outside of tauri) so he does not have that much time to work on it right now. Unfortunetely the changes (in both tauri and egui) are so large that it looks like the crates are basically getting a full rewrite.

I can't give any ETA whatsoever at this point, work on it _may_ not resume before tauri v2 hits rc or stable.

**Reply by samkearney** [reply]
Totally understandable, I know how work like this can stretch people thin. Thank you for the response.

**Comment by clearlysid** [comment]
Hey folks! Some good news → I've gotten a basic version of this [working here](https://github.com/clearlysid/egui-tao) along with some thoughts on how we can perhaps offer a way forward in the README.

I will be happy to accelerate my work on this if there's some community interest.

**Reply by ShayBox** [reply]
Great Work! I hope this can be used to update tauri-egui.

**Reply by jason-ni** [reply]
> Hey folks! Some good news → I've gotten a basic version of this [working here](https://github.com/clearlysid/egui-tao) along with some thoughts on how we can perhaps offer a way forward in the README.
> 
> I will be happy to accelerate my work on this if there's some community interest.

Thanks for your great idea! I [made a fork ](https://github.com/jason-ni/tauri-plugin-egui)and extended it with "multi-window" and "on-destroy" support.  And my goal is to create a screenshot app like snapaste. Everything looks good until I check the performance monitor. There is memory leak.  I infer that the leak is caused by the rendering component. Because creating and closing simple empty egui windows doesn't cause leak. 

I also tried to use glow to replace the wgpu render. You can check the "glow" branch in my fork. It works but also has memory leak.

Any guys also have love for Tauri & Egui can have a try. It's allmost get what we want for a long time: when you want heavy machine you got webview, when you want light knife you got egui, and they are all in the light pacaged Tauri distribution.

**Reply by clearlysid** [reply]
@jason-ni this is great work! Will be happy to merge it if we can solve the memory leak 👏

**Reply by jason-ni** [reply]
I spent more time on this and still can't find the root cause of this memory leak. What's more, I find that the Iced GUI multi_window example also has similar memory leak pattern in my Macbook. https://github.com/iced-rs/iced/issues/3233

**Reply by jason-ni** [reply]
Now I have further progress on this path: https://github.com/jason-ni/tauri-plugin-iced

I elaborated the leakage behavior in the README of this project. Now if we take Tauri and winit both as black boxies, we can make conclution that Tauri raw window create/close leak memory in MacOS. The leakage is related to graphic content, as the leak in an empty window isn't obvious, but the leak in a window with images is quite noticeable. 

@FabianLars how do you think current plugin solution that integrate Egui and Iced to wry eventloop? It's much cleaner comparing to the original one that requires several upstream hack. If we solve the memory leak issue, this solution can help expand Tauri's use cases.
